imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java

```diff
@@ -0,0 +1,1222 @@
+                    mSnapBackTime = fakeStartPos;
+                    mediaConfig.setStartPosition(0L);
+                } else {
+                    mediaConfig.setStartPosition((long) fakeStartPos);
+                }
+            }
+        }
+    }
+
+    @Override
+    protected void onUpdateConfig(Object config) {
+        log.d("Start onUpdateConfig");
+        adConfig = parseConfig(config);
+    }
+
+    @Override
+    protected void onApplicationPaused() {
+        appIsInBackground = true;
+        if (isAdDisplayed) {
+            isAdIsPaused = true;
+        }
+    }
+
+    @Override
+    protected void onApplicationResumed() {
+        appIsInBackground = false;
+        if (shouldPrepareOnResume) {
+            if (isAutoPlay) {
+                player.prepare(mediaConfig);
+                player.play();
+            } else {
+                player.prepare(mediaConfig);
+            }
+            shouldPrepareOnResume = false;
+            return;
+        }
+        if (getPlayerEngine() != null && isAdShouldAutoPlayOnResume() && isAdDisplayed) {
+            getPlayerEngine().play();
+        }
+    }
+
+    @Override
+    protected void onDestroy() {
+        isAdRequested = false;
+        clearAdsLoader();
+        resetIMA();
+    }
+
+    @Override
+    public void onAdEvent(com.google.ads.interactivemedia.v3.api.AdEvent adEvent) {
+        if (streamManager == null) {
+            log.w("WARNING, streamManager == null");
+            return;
+        }
+
+        if(adEventsMap == null){
+            log.e("ERROR, adEventsMap == null");
+            return;
+        }
+
+        lastAdEventReceived = adEventsMap.get(adEvent.getType());
+        if (adEvent.getType() != com.google.ads.interactivemedia.v3.api.AdEvent.AdEventType.AD_PROGRESS) {
+            log.d("Event: " + adEvent.getType());
+        }
+        switch (adEvent.getType()) {
+            case AD_BREAK_STARTED: //Fired when an ad break starts.
+                log.d("AD AD_BREAK_STARTED");
+                isAdDisplayed = true;
+                messageBus.post(new AdEvent(AdEvent.Type.AD_BREAK_STARTED));
+                messageBus.post(new AdEvent(AdEvent.Type.CONTENT_PAUSE_REQUESTED));
+                break;
+            case AD_BREAK_ENDED: //Fired when an ad break ends.
+                log.d("AD AD_BREAK_ENDED");
+                messageBus.post(new AdEvent(AdEvent.Type.AD_BREAK_ENDED));
+                messageBus.post(new AdEvent(AdEvent.Type.CONTENT_RESUME_REQUESTED));
+                boolean allAdsPlayed = true;
+                boolean lastAdPlayed = true;
+                List<CuePoint> cuesList = streamManager.getCuePoints();
+                if (cuesList != null) {
+                    for (int i = 0 ; i < cuesList.size() ; i++) {
+                        if (!cuesList.get(i).isPlayed()) {
+                            allAdsPlayed = false;
+                        }
+                        if (i == cuesList.size() -1) {
```

Comment:
Use iterator.hasNext() to detect the last item in the list.

Reply:
done